lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -510,6 +510,9 @@ def add_address(post, options)
       end
 
       def add_billing_address(post, options, address)
+        address[:address1] = 'NA' if address[:address1].nil? || address[:address1].blank?
+        address[:address2] = 'NA' if address[:address2].nil? || address[:address2].blank?
```

Comment:
Based on [this description](https://stackoverflow.com/a/888877) of the behavior of `.blank?`, I think the `.nil?` is extraneous here - `.blank?` should return true for both `nil` and empty strings.

I think you could simply swap the `nil?` for `blank?` and then also remove the `|| 'NA'` fallback at the end of the `address_override` ternary statements, because it won't be needed anymore (and removing it will make the intention of that logic a little easier to parse)

Reply:
sounds good!

Reply:
good to know .blank? also accounts for nil values!